shaders/shaderDensityShared.comp

```diff
@@ -0,0 +1,83 @@
+#version 310 es
+precision mediump float;
+layout(local_size_x = 256, local_size_y = 1, local_size_z = 1) in;
+
+void GroupMemoryBarrierWithGroupSync()
+{
+    groupMemoryBarrier();
```

Comment:
Should use `memoryBarrierShared ()` instead of `groupMemoryBarrier()`. Please modify all similar places.